bus-mapping/src/evm/opcodes/extcodehash.rs

```diff
@@ -0,0 +1,295 @@
+use super::Opcode;
+use crate::{
+    circuit_input_builder::CircuitInputStateRef,
+    operation::{
+        AccountField, AccountOp, CallContextField, CallContextOp, TxAccessListAccountOp, RW,
+    },
+    state_db::Account,
+    Error,
+};
+use eth_types::{evm_types::GasCost, GethExecStep, ToAddress, ToWord};
+
+#[derive(Debug, Copy, Clone)]
+pub(crate) struct Extcodehash;
+
+impl Opcode for Extcodehash {
+    fn gen_associated_ops(
+        state: &mut CircuitInputStateRef,
+        steps: &[GethExecStep],
+    ) -> Result<(), Error> {
+        let step = &steps[0];
+        let stack_address = step.stack.last_filled();
+
+        // Pop external address off stack
+        let external_address = step.stack.last()?.to_address();
+        state.push_stack_op(RW::READ, stack_address, external_address.to_word())?;
+
+        // Read transaction id from call context
+        state.push_op(
+            RW::READ,
+            CallContextOp {
+                call_id: state.call()?.call_id,
+                field: CallContextField::TxId,
+                value: state.tx_ctx.id().into(),
+            },
+        );
+
+        // Update transaction access list for external_address
+        let account_previously_accessed = match step.gas_cost {
+            GasCost::WARM_STORAGE_READ_COST => true,
+            GasCost::COLD_ACCOUNT_ACCESS_COST => false,
+            _ => unreachable!(),
+        };
```

Comment:
We want to track the access list state in the StateDB when processing each step.  You can do that like this:
```rust
state.sdb.add_account_to_access_list(external_address);
```
See the gen_associated_ops of the begin_tx for reference: https://github.com/appliedzkp/zkevm-circuits/blob/c5dda20e901951e61f751b89dd50607f83b98a95/bus-mapping/src/evm/opcodes.rs#L270